src/Matrix.php

```diff
@@ -957,6 +957,110 @@ public function matmul(Matrix $b) : self
         return self::quick($c);
     }
 
+    /**
+     * Multiply this matrix with another matrix using the Strassen algorithm (matrix-matrix product).
+     *
+     * @param \Tensor\Matrix $b
+     * @throws \InvalidArgumentException
+     * @return self
+     */
+    public function matmulStrassen(Matrix $b) : self
```

Comment:
Depending on the data we collect from the benchmark, we may be able to take this method private and double dispatch to this method from `matmul()` if the matrix shape falls within the range that Strassen is faster than the O(n^3) method for. But we'll know more when we start experimenting.